### PR TITLE
[SRVKS-1088] Add the "Configuring burst and QPS for net-kourier" docs

### PR DESCRIPTION
**Pull request by rh-max**
Version(s):
Serverless 1.29+ (CP to branches `serverless-docs-1.29` and `serverless-docs-1.30`)

Issue:
https://issues.redhat.com/browse/SRVKS-1088

Link to docs preview:
https://62138--docspreview.netlify.app/openshift-serverless/latest/knative-serving/kube-burst-qps-net-kourier.html

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

File: modules/serverless-configuring-burst-qps-for-net-kourier.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+//
+// * knative-serving/kube-burst-qps-net-kourier.adoc
+
+:_content-type: REFERENCE
+[id="serverless-configuring-burst-qps-for-net-kourier_{context}"]
+= Configuring burst and QPS values for net-kourier
+
+The queries per second (QPS) value determines the number of client requests or API calls that are sent to the API server.
+
+The burst value determines how many requests the client can make in excess of the QPS rate. This is helpful for controllers that are bursty in nature and do not spread their requests uniformly in time.
```
Comment (mgencur) [inline modules/serverless-configuring-burst-qps-for-net-kourier.adoc:11]: As I read this first sentence it's not fully clear to me how "burst" works. I think we could make it more clear in this way: `The burst value determines how many requests from the client can be stored for processing. Requests exceeding this buffer will be dropped.`

File: modules/serverless-configuring-burst-qps-for-net-kourier.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    - container: controller
+      envVars:
+      - name: KUBE_API_BURST
+        value: "200" <1>
+      - name: KUBE_API_QPS
+        value: "200" <2>
+----
++
+<1> The QPS rate of communication between controller and the API Server. The default value is 200.
+<2> .. The burst rate of communication between Kubelet and the API Server. The default value is 200.
```
Comment (mgencur) [inline modules/serverless-configuring-burst-qps-for-net-kourier.adoc:38]:
```suggestion
<2> .. The burst capacity of communication between Kubelet and the API Server. The default value is 200.
```

Comment (mgencur) [reply modules/serverless-configuring-burst-qps-for-net-kourier.adoc:38]: "burst" is technically not a "rate".

**Review by mgencur — COMMENTED** [review]
Just some minor suggestions. Otherwise looks good.

**Comment by rh-max** [comment]
@mgencur Thank you, Martin! Implemented the suggestions.

**Comment by mgencur** [comment]
@rh-max Thanks.

LGTM from QE

File: modules/serverless-configuring-burst-qps-for-net-kourier.adoc [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  - name: net-kourier-controller
+    env:
+    - container: controller
+      envVars:
+      - name: KUBE_API_BURST
+        value: "200" <1>
+      - name: KUBE_API_QPS
+        value: "200" <2>
+----
++
```
Comment (kelbrown20) [inline modules/serverless-configuring-burst-qps-for-net-kourier.adoc:36]: You can see at the bottom of the [preview](https://62138--docspreview.netlify.app/openshift-serverless/latest/knative-serving/kube-burst-qps-net-kourier.html#serverless-configuring-burst-qps-for-net-kourier_kube-burst-qps-net-kourier) that the callouts are not rendering properly. I tested it locally and removing the `+` should fix the issue
```suggestion

```

File: modules/serverless-configuring-burst-qps-for-net-kourier.adoc [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+//
+// * knative-serving/kube-burst-qps-net-kourier.adoc
+
+:_content-type: REFERENCE
+[id="serverless-configuring-burst-qps-for-net-kourier_{context}"]
+= Configuring burst and QPS values for net-kourier
+
+The queries per second (QPS) value determines the number of client requests or API calls that are sent to the API server.
+
+The burst value determines how many requests from the client can be stored for processing. Requests exceeding this buffer will be dropped. This is helpful for controllers that are bursty in nature and do not spread their requests uniformly in time.
```
Comment (kelbrown20) [inline modules/serverless-configuring-burst-qps-for-net-kourier.adoc:11]:
```suggestion
The burst value determines how many requests from the client can be stored for processing. Requests exceeding this buffer will be dropped. This is helpful for controllers that are bursty and do not spread their requests uniformly in time.
```

File: modules/serverless-configuring-burst-qps-for-net-kourier.adoc [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+When the `net-kourier-controller` restarts, it parses all `ingress` resources deployed on the cluster, which leads to a significant number of API calls. Consequently, it takes a long time for `net-kourier-controller` to start.
```
Comment (kelbrown20) [inline modules/serverless-configuring-burst-qps-for-net-kourier.adoc:13]:
```suggestion
When the `net-kourier-controller` restarts, it parses all `ingress` resources deployed on the cluster, which leads to a significant number of API calls. Due to this, the `net-kourier-controller` can take a long time to start.
```

File: modules/serverless-configuring-burst-qps-for-net-kourier.adoc [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+You can adjust the QPS and burst values for `net-kourier-controller` in the KnativeServing CR:
```
Comment (kelbrown20) [inline modules/serverless-configuring-burst-qps-for-net-kourier.adoc:15]:
```suggestion
You can adjust the QPS and burst values for the `net-kourier-controller` in the KnativeServing CR:
```

**Review by kelbrown20 — COMMENTED** [review]
Hello! There is just a small format fix that needs updating, and some working suggestions. Great work on this!

**Comment by rh-max** [comment]
> Hello! There is just a small format fix that needs updating, and some working suggestions. Great work on this!

Hi Kelly, thank you, I applied your suggestions.

File: modules/serverless-configuring-burst-qps-for-net-kourier.adoc [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    env:
+    - container: controller
+      envVars:
+      - name: KUBE_API_BURST
+        value: "200" <1>
+      - name: KUBE_API_QPS
+        value: "200" <2>
+----
+<1> The QPS rate of communication between controller and the API Server. The default value is 200.
+<2> .. The burst capacity of communication between Kubelet and the API Server. The default value is 200.
```
Comment (jldohmann) [inline modules/serverless-configuring-burst-qps-for-net-kourier.adoc:37]: Should the `..` be here? It looks out of place.

**Review by jldohmann — COMMENTED** [review]
Minor nit/question before merge, otherwise LGTM. Please ping me again and I'll be happy to merge, thank you!

File: modules/serverless-configuring-burst-qps-for-net-kourier.adoc (hunk 7 above)
Comment (rh-max) [reply modules/serverless-configuring-burst-qps-for-net-kourier.adoc]: Thanks Jesse, it is fixed now.

**Review by jldohmann — APPROVED** [review]

**Pull request merged**